In [1]:
import pandas as pd

#Pandas выводит 20 стобцов и 60 строк, воспользуемся set_option и выведем столбцы по максимуму 
pd.set_option('display.max_columns', None)

In [2]:
# Предварительная обработка данных
#Считайте таблицу с признаками из файла features.csv с помощью кода, приведенного выше.
#features = pandas.read_csv('./features.csv', index_col='match_id')

data = pd.read_csv('features.csv', index_col='match_id')

In [3]:
#Выведем полученные статистические данные 
stat = data.describe()


Описание признаков в таблице

    match_id: идентификатор матча в наборе данных
    start_time: время начала матча (unixtime)
    lobby_type: тип комнаты, в которой собираются игроки (расшифровка в dictionaries/lobbies.csv)
    Наборы признаков для каждого игрока (игроки команды Radiant — префикс rN, Dire — dN):
        r1_hero: герой игрока (расшифровка в dictionaries/heroes.csv)
        r1_level: максимальный достигнутый уровень героя (за первые 5 игровых минут)
        r1_xp: максимальный полученный опыт
        r1_gold: достигнутая ценность героя
        r1_lh: число убитых юнитов
        r1_kills: число убитых игроков
        r1_deaths: число смертей героя
        r1_items: число купленных предметов
    Признаки события "первая кровь" (first blood). Если событие "первая кровь" не успело произойти за первые 5 минут, то признаки принимают пропущенное значение
        first_blood_time: игровое время первой крови
        first_blood_team: команда, совершившая первую кровь (0 — Radiant, 1 — Dire)
        first_blood_player1: игрок, причастный к событию
        first_blood_player2: второй игрок, причастный к событию
    Признаки для каждой команды (префиксы radiant_ и dire_)
        radiant_bottle_time: время первого приобретения командой предмета "bottle"
        radiant_courier_time: время приобретения предмета "courier"
        radiant_flying_courier_time: время приобретения предмета "flying_courier"
        radiant_tpscroll_count: число предметов "tpscroll" за первые 5 минут
        radiant_boots_count: число предметов "boots"
        radiant_ward_observer_count: число предметов "ward_observer"
        radiant_ward_sentry_count: число предметов "ward_sentry"
        radiant_first_ward_time: время установки командой первого "наблюдателя", т.е. предмета, который позволяет видеть часть игрового поля
    Итог матча (данные поля отсутствуют в тестовой выборке, поскольку содержат информацию, выходящую за пределы первых 5 минут матча)
        duration: длительность
        radiant_win: 1, если победила команда Radiant, 0 — иначе
        Состояние башен и барраков к концу матча (см. описание полей набора данных)
            tower_status_radiant
            tower_status_dire
            barracks_status_radiant
            barracks_status_dire

In [17]:
#Проверка пропусков в данных

rows = len(data)#длина 97230.0
count = stat.T['count'] #преобразуем строку 'count' в столбец
count_NAN = count[count < rows]#все меньше 97230.0
#print count_NAN
#Переведем в % для удобства анализа
count_NAN.sort_values().apply(lambda i: ((rows - i) / rows)*100)

first_blood_player2            45.240152
radiant_flying_courier_time    28.261853
dire_flying_courier_time       26.841510
first_blood_time               20.110048
first_blood_team               20.110048
first_blood_player1            20.110048
dire_bottle_time               16.602900
radiant_bottle_time            16.138023
radiant_first_ward_time         1.888306
dire_first_ward_time            1.878021
radiant_courier_time            0.711714
dire_courier_time               0.695259
Name: count, dtype: float64

 Много ли пропусков в данных? Запишите названия признаков, имеющих пропуски, и попробуйте для любых двух из них дать обоснование, почему их значения могут быть пропущены.
 
Обоснование: 
 Признаки события "первая кровь" (first blood). Если событие "первая кровь" не успело произойти за первые 5 минут, то признаки принимают пропущенное значение
 
 first_blood_player2  - второй игрок не участвовал в событии "первое убийство" в 45% случаев. Видимо, первый игрок обошелся своими силами и помощь не потребовалась.
  
 first_blood_time  - в 20% случаев "первой крови" еще нет в первые 5 минут игры
 first_blood_team  - в 20% случаев "первой крови" еще нет в первые 5 минут игры
 first_blood_player1 - тоже самое, что и в двух предыдущих
 

